src/arkreactor/Compiler/Compiler.cpp

```diff
@@ -719,8 +719,33 @@ namespace Ark
 
             auto it = std::find(m_defined_symbols.begin(), m_defined_symbols.end(), str);
             if (it == m_defined_symbols.end() && !is_plugin)
-                throwCompilerError("Unbound variable error (variable is used but not defined)", sym);
+            {
+                std::string suggestion = offerSuggestion(str);
+                if (suggestion.empty())
+                    throwCompilerError("Unbound variable error \"" + str + "\" (variable is used but not defined)", sym);
+
+                throwCompilerError("Unbound variable error \"" + str + "\" (did you mean \"" + suggestion + "\"?)", sym);
+            }
+        }
+    }
+
+    std::string Compiler::offerSuggestion(const std::string& str)
+    {
+        std::string suggestion;
+        // our suggestion shouldn't require more than half the string to change
+        int suggestion_distance = str.size() / 2;
```

Comment:
Should be a `std::size_t` here as well

Reply:
This has been changed to be a `std::size_t`. I've also changed `current_distance` to be a `std::size_t` to avoid compiler warnings about comparing integers of different types.